internal/envoy/v3/route.go

```diff
@@ -233,6 +233,12 @@ func routeRedirect(redirect *dag.Redirect) *envoy_route_v3.Route_Redirect {
 		r.Redirect.PortRedirect = redirect.PortNumber
 	}
 
+	if len(redirect.Path) > 0 {
+		r.Redirect.PathRewriteSpecifier = &envoy_route_v3.RedirectAction_PrefixRewrite{
```

Comment:
should this be `RedirectAction_PathRedirect` instead of `RedirectAction_PrefixRewrite`?

Reply:
I'm not 100% sure to be honest. They both work the same way short of path_redirect allows for stripping the query params.

Reply:
I would think `prefix_rewrite` would replace a prefix of the path component, whereas `path_redirect` would replace the entire path component with the specified value? So I'd guess that in your E2E test right now, if you made a request to `/path-rewrite/foo`, then the redirect would send you to `/blog/foo`?

Valid use cases for both, but we should be clear on which we're implementing.

Reply:
Ahh ok great point, I overlooked that. I'll fix it up. We should just add the others as well for when folks need it, but will do that in a future PR.

Reply:
yeah im curious to see what the differences between the two end up being, esp. since currently there are only prefix path matches allowed in httpproxy, gonna try it out

Reply:
any chance we want to support both? could make the path redirect option a struct to future proof or tack on a bool later